### PR TITLE
Bugfix/kvstore charge type

### PR DESCRIPTION
**Pull request by jamespan**
When change `instance_charge_type`, provider would newly create a kvstore instance, which is not expected.

This PR fix this issue by change the charge_type of the kvstore instance.

### CODE REVIEW

File: alicloud/resource_alicloud_kvstore_instance.go [hunk 1]
```diff
@@ -49,7 +50,7 @@ func resourceAlicloudKVStoreInstance() *schema.Resource {
 				Type:         schema.TypeString,
 				ValidateFunc: validateInstanceChargeType,
 				Optional:     true,
-				ForceNew:     true,
+				ForceNew:     false,
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_kvstore_instance.go]: Just removing `ForceNew: true` is ok. And then modify the docs.

File: alicloud/resource_alicloud_kvstore_instance.go [hunk 2]
```diff
@@ -206,6 +209,24 @@ func resourceAlicloudKVStoreInstanceUpdate(d *schema.ResourceData, meta interfac
 		d.SetPartial("password")
 	}
 
+	if d.HasChange("instance_charge_type") {
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_kvstore_instance.go]: I think line 212 and 221 can be merged into one block.

File: alicloud/resource_alicloud_kvstore_instance.go [hunk 3]
```diff
@@ -206,6 +209,24 @@ func resourceAlicloudKVStoreInstanceUpdate(d *schema.ResourceData, meta interfac
...
+		prePaidRequest.InstanceId = d.Id()
+		prePaidRequest.Period = requests.Integer(strconv.Itoa(d.Get("period").(int)))
+		update = true
+		changePayType = true
+		d.SetPartial("instance_charge_type")
+		d.SetPartial("period")
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_kvstore_instance.go]: Line 217-218 should be set after invoking api successfully.

File: alicloud/resource_alicloud_kvstore_instance.go [hunk 4]
```diff
@@ -221,6 +242,21 @@ func resourceAlicloudKVStoreInstanceUpdate(d *schema.ResourceData, meta interfac
...
+				return rkvClient.TransformToPrePaid(prePaidRequest)
+			})
+		}
+		if err != nil {
+			return fmt.Errorf("TransformToPrePaid got an error: %#v", err)
+		}
+		// wait instance status is Normal after modifying
+		if err := kvstoreService.WaitForRKVInstance(d.Id(), Normal, DefaultLongTimeout); err != nil {
+			return fmt.Errorf("WaitForInstance %s got error: %#v", Normal, err)
+		}
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_kvstore_instance.go]: I think line 245-259 should not be added into `update` block.

**Comment by jamespan** [comment]
@xiaozhu36 I pushed a new commit, fixed all issues found by your latest review

**Pull request merged**